aea/helpers/profiler_type_black_list.py

```diff
@@ -0,0 +1,157 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2022 Valory AG
+#   Copyright 2018-2021 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""Profiler blacklisted types."""
+
+PROFILER_TYPE_BLACK_LIST = [
```

Comment:
Why did you chose the black list approach instead of a whitelist one?

Reply:
Two reasons:
- AFAIK from open-aea we don't have direct access to objects defined in consensus-algorithms. I don't know if we could import them into the profiler, but doing could create circular dependencies I guess. So in order to see those, instead of importing them and whitelisting them (which needs importing), I went the reverse route.

- The blacklisted objects are not likely to bring  valuable information. Basically I looked at the gc in an empty script and saw that there were already thousands and hundreds of these blacklisted objects.So even if we see thousands of them in our aea executions, that information won't be very useful and will mostly hide the objects in our library that grow.

Reply:
I see. I'm not sure how reliable this blacklist is. Sounds pretty ad hoc.

Reply:
Yes, it is. But if we want to track the whole gc, either we show hundreds of objects in the profiler output until we see the ones from our stack, or we filter them somehow. Other option is to let the dev decide the objects of interest, or drop the gc scanning entirely. The downside is that we would lose the ability of tracking objects outside of open-aea (like consensus).

Reply:
The alternative would be to let the user specify to-be-tracked objects in the config file and then using that to import them and tracking them. But that's more involved and for now your approach will do.